testing/fpgad/test_command_line_c.cpp

```diff
@@ -234,5 +244,179 @@ TEST_P(fpgad_command_line_c_p, canonicalize1) {
   config_.daemon = true;
 }
 
+/**
+ * @test       canonicalize2
+ * @brief      Test: cmd_canonicalize_paths
+ * @details    If the .cfgfile member of<br>
+ *             the input config contains "..",<br>
+ *             then the fn falls back to searching<br>
+ *             for a config file.<br>
+ */
+TEST_P(fpgad_command_line_c_p, canonicalize2) {
+  strcpy(config_.cfgfile, "../some.cfg");
+
+  cmd_canonicalize_paths(&config_);
+
+  EXPECT_EQ(config_.cfgfile[0], '\0');
+}
+
+/**
+ * @test       canonicalize3
+ * @brief      Test: cmd_canonicalize_paths
+ * @details    If the .cfgfile member of<br>
+ *             the input config is a valid file,<br>
+ *             but it has invalid configuration contents,<br>
+ *             then the fn attempts to use it to load<br>
+ *             the configuration, but fails over to searching.<br>
+ */
+TEST_P(fpgad_command_line_c_p, canonicalize3) {
+  strcpy(config_.cfgfile, cfg_file_);
+
+  cmd_canonicalize_paths(&config_);
+
+  EXPECT_STREQ(config_.cfgfile, "");
+}
+
+/**
+ * @test       canonicalize4
+ * @brief      Test: cmd_canonicalize_paths
+ * @details    If the .cfgfile member of<br>
+ *             the input config is a valid file,<br>
+ *             and it has valid configuration contents,<br>
+ *             then the fn attempts to use it to load<br>
+ *             the configuration and succeeds.<br>
+ */
+TEST_P(fpgad_command_line_c_p, canonicalize4) {
+  const char *valid_cfg = R"cfg(
+{
+  "configurations": {
+    "a": {
+      "configuration": {},
+      "enabled": true,
+      "plugin": "liba.so",
+      "devices": [
+        [ "0x8086", "0x0b30" ]
+      ]
+    }
+  },
+  "plugins": [
+    "a"
+  ]
+}
+)cfg";
+
+  strcpy(config_.cfgfile, cfg_file_);
+  std::ofstream cfg;
+  cfg.open(cfg_file_, std::ios::out);
+  cfg.write(valid_cfg, strlen(valid_cfg));
+  cfg.close();
+
+  cmd_canonicalize_paths(&config_);
+
+  char *d = get_current_dir_name();
```

Comment:
Would it be a good idea to do an assert on d != NULL before continuing?

Reply:
:+1: